src/backend/distributed/transaction/backend_data.c

```diff
@@ -408,7 +404,6 @@ StoreAllActiveTransactions(Tuplestorestate *tupleStore, TupleDesc tupleDescripto
 
 		Oid databaseId = currentBackend->databaseId;
 		int backendPid = ProcGlobal->allProcs[backendIndex].pid;
-		initiatorNodeIdentifier = currentBackend->citusBackend.initiatorNodeIdentifier;
```

Comment:
These were in the order of the `values` below. Can you move `nodeId` assignment to here

Reply:
We cannot call `ExtractNodeIdFromGlobalPID`  before `SpinLockRelease`. 

We need currentBackend->globalPID as a param for `ExtractNodeIdFromGlobalPID`, so I don't think it is worth doing one more `Acquire/Release` spinlock to keep the order match?

Reply:
Fair enough